arduino/smartcar/smartcar.ino

```diff
@@ -155,7 +165,69 @@ void loop()
     
 }
 
-//If the connection breaks, this method will be called
+
+void stopZoneAutoBreak(long frontUltDis, long frontIRDis, long backIRDis)
+{
+     if (frontUltDis <= 100 && frontUltDis != 0 || frontIRDis <= 40 && frontIRDis != 0 || backIRDis <= 40 && backIRDis != 0)//stop zone
+         {
+            if (canDrive)//check whether you're in the stop zone
+            {
+                car.setSpeed(0);
+                Serial.println("Emergency stop 1");
+            }
+            canDrive = false;//so the car can move in the stop soon
+        } else {
+            canDrive = true;//so the car will stop again if it hits the stop zone
+        }
+}
+
+//Threshold means to close to an object/ different thresholds means level of closeness
+void incomingAvoidanceThreshold(long frontUltDis, long frontIRDis, long backIRDis)
+{
+    if (frontUltDis <= 30 && frontUltDis != 0 || frontIRDis <= 40 && frontIRDis != 0)//Ford threshold 1
```

Comment:
Add more detail in your comment. I do not understand what "ford threshold 1" means